geowebcache/diskquota/core/src/main/java/org/geowebcache/diskquota/UsageStatsMonitor.java

```diff
@@ -82,24 +82,32 @@ public void shutDown() {
         final int maxAttempts = 6;
         final int seconds = 5;
         int attempts = 1;
-        while (!executorService.isTerminated()) {
-            attempts++;
-            try {
-                awaitTermination(seconds, TimeUnit.SECONDS);
-            } catch (InterruptedException e) {
-                String message =
-                        "Usage statistics thread helper for DiskQuota failed to shutdown within "
-                                + (attempts * seconds)
-                                + " seconds. Attempt "
-                                + attempts
-                                + " of "
-                                + maxAttempts
-                                + "...";
-                log.warn(message);
-                if (attempts == maxAttempts) {
-                    throw new RuntimeException(message, e);
+        boolean interrupted = false;
+        try {
+            while (!executorService.isTerminated()) {
+                attempts++;
+                try {
+                    awaitTermination(seconds, TimeUnit.SECONDS);
+                } catch (InterruptedException e) {
+                    interrupted = true;
+                    String message =
+                            "Usage statistics thread helper for DiskQuota failed to shutdown within "
+                                    + (attempts * seconds)
+                                    + " seconds. Attempt "
+                                    + attempts
+                                    + " of "
+                                    + maxAttempts
+                                    + "...";
+                    log.warn(message);
+                    if (attempts == maxAttempts) {
+                        throw new RuntimeException(message, e);
+                    }
```

Comment:
This whole block seems entirely identical to the one in QuotaUpdatesMonitor, above. Can it be moved to a shared superclass?
If not then perhaps update the error message to differentiate the two?

Reply:
Refactored to put most of the duplicate code in AbstractMonitor